src/main/java/org/opensearch/jobscheduler/utils/JobDetailsService.java

```diff
@@ -33,31 +33,89 @@
+public class JobDetailsService implements IndexingOperationListener {
 
     private static final Logger logger = LogManager.getLogger(JobDetailsService.class);
-    private static final String JOB_DETAILS_INDEX_NAME = ".opensearch-plugins-job-details";
+    public static final String JOB_DETAILS_INDEX_NAME = ".opensearch-plugins-job-details";
     private static final String PLUGINS_JOB_DETAILS_MAPPING_FILE = "/mappings/opensearch_plugins_job_details.json";
 
     public static Long TIME_OUT_FOR_REQUEST = 10L;
     private final Client client;
     private final ClusterService clusterService;
+    private Set<String> indicesToListen;
 
-    public JobDetailsService(final Client client, final ClusterService clusterService) {
+    private static final ConcurrentMap<String, JobDetails> indexToJobDetails = IndexToJobDetails.getInstance();
+
+    public JobDetailsService(final Client client, final ClusterService clusterService, Set<String> indicesToListen) {
         this.client = client;
         this.clusterService = clusterService;
+        this.indicesToListen = indicesToListen;
     }
 
     public boolean jobDetailsIndexExist() {
         return clusterService.state().routingTable().hasIndex(JOB_DETAILS_INDEX_NAME);
     }
 
+    public static ConcurrentMap<String, JobDetails> getIndexToJobDetails() {
+        return JobDetailsService.indexToJobDetails;
+    }
+
+    private void updateIndicesToListen(String jobIndex) {
+        this.indicesToListen.add(jobIndex);
+    }
+
+    /**
+     * Registers or updates a jobDetails entry
+     *
+     * @param extensionId the unique Id for the job details
+     * @param jobDetails the jobDetails to register
+     */
+    void updateIndexToJobDetails(String extensionId, JobDetails jobDetails) {
+        if (indexToJobDetails.containsKey(extensionId)) {
+            if (jobDetails.getJobType() != null) {
+                // Update JobDetails entry with job type
+                JobDetails existingJobDetails = indexToJobDetails.get(extensionId);
+                existingJobDetails.setJobType(jobDetails.getJobType());
+            }
+        } else {
+            // Register JobDetails entry
+            indexToJobDetails.put(extensionId, jobDetails);
+            updateIndicesToListen(jobDetails.getJobIndex());
+        }
+    }
+
+    @Override
+    public void postIndex(ShardId shardId, Engine.Index index, Engine.IndexResult result) {
+
+        // Determine if index operation was successful
+        if (result.getResultType().equals(Engine.Result.Type.FAILURE)) {
+            logger.info("Job Details Registration failed for extension {} on index {}", index.id(), shardId.getIndexName());
+            return;
+        }
+
+        // Generate parser using bytesRef from index
+        try {
+            XContentParser parser = XContentType.JSON.xContent()
+                .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, index.source().utf8ToString());
```

Comment:
Do we really not need any XContent from anywhere in this plugin or OpenSearch?  OK if so, just had to ask.

Reply:
No the XContentRegistry is not needed here to read the `JobDetails` object from the index. The only NamedXcontentRegistry entry that Job Scheduler registers is the parser for `Schedule` [here](https://github.com/opensearch-project/job-scheduler/blob/6f93b49c88df3476a786216e190702d853bbe69f/src/main/java/org/opensearch/jobscheduler/JobSchedulerPlugin.java#LL179C2-L179C2), which is not used within the `JobDetailsService`